acceptance/tests/windows/stop_puppet_svc_stops_puppet.rb

```diff
@@ -0,0 +1,44 @@
+test_name '(C14941) Stopping puppet service on Windows stops puppet'
+
+confine :to, :platform => 'windows'
+
+opts = {
+  :max_retries => 2,
+  :retry_interval => 5,
+}
+
+agents.each do
+  step 'Make sure puppet service is running' do
+    query = on(agent, 'sc query puppet', :acceptable_exit_codes => [0])
+    if ! query.stdout.include?('RUNNING')
+      start = retry_on(agent, 'sc start puppet', opts)
+      fail_test 'puppet service failed to start' \
+        unless start.stdout.include?('RUNNING')
+    end
+  end
+ 
+  step 'Stop the puppet service and verify it stopped' do
+    stop = on(agent, 'sc stop puppet', :acceptable_exit_codes => [0])
+    if ! stop.stdout.include?('STOPPED')
+      query = retry_on(agent, 'sc query puppet', opts)
+      fail_test 'puppet service failed to stop' \
+        unless query.stdout.include?('STOPPED')
+    end
+  end
+
+  step 'Restart the puppet service and verify it restarted' do
+    restart = on(agent, 'sc start puppet', :acceptable_exit_codes => [0])
+    if ! restart.stdout.include?('RUNNING')
+      ### FIXME: Doesn't pass without the sleep; the query command succeeds
+      ### so retry_on returns before we get the status we want. The 
+      ### sc command will succeed while the service status remains
+      ### START_PENDING. As a result, the test fails. The hard sleep 
+      ### gives the service "enough" time to start
+      sleep 5
```

Comment:
https://github.com/puppetlabs/beaker/blob/master/lib/beaker/shared/repetition.rb#L14

not sure if that helps your success conditions (might need better options for the various `repeat` methods in beaker).